File: 1544-make-the-string-great/1544-make-the-string-great.cpp
class Solution {
public:
    string makeGood(string s) {
        stack<char> st;
        string str="";
        int size=s.size();
        for(int i=0;i<size;i++)
        {
            if(st.empty()==0)
            {
                if(st.top()==s[i]+32||st.top()==s[i]-32)
                    st.pop();
                else
                {
                    st.push(s[i]);
                }
            }
            else 
            {
                st.push(s[i]);
            }
        }
        while(st.empty()==0)
        {
            str=str+st.top();
            st.pop();
        }
        reverse(str.begin(),str.end());
        return str;
    }
};

File: 1544-make-the-string-great/1544-make-the-string-great.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack=[]
        for ch in s:
            if stack and stack[-1]!=ch and stack[-1].upper()==ch.upper():
                stack.pop()
            else:
                stack.append(ch)
        return "".join(stack)
